mplib/planner.py

```diff
@@ -378,6 +378,41 @@ def TOPP(self, path, step=0.1, verbose=False):
         qdds_sample = jnt_traj(ts_sample, 2)
         return ts_sample, qs_sample, qds_sample, qdds_sample, jnt_traj.duration  # type: ignore
 
+    def TOPP_SD(self, path, duration, step=0.1, verbose=False):
+        """
+        Path Parameterization with desired duration (in seconds)
+        Args:
+            path: numpy array of shape (n, dof)
+            duration: desired duration of the path in seconds
+            step: step size for the discretization
+            verbose: if True, will print the log of TOPPRA
+        """
+
+        N_samples = path.shape[0]
+        dof = path.shape[1]
+        assert dof == len(self.joint_vel_limits)
+        assert dof == len(self.joint_acc_limits)
+        ss = np.linspace(0, 1, N_samples)
+        path = ta.SplineInterpolator(ss, path)
+        pc_vel = constraint.JointVelocityConstraint(self.joint_vel_limits)
+        pc_acc = constraint.JointAccelerationConstraint(self.joint_acc_limits)
+        instance = algo.TOPPRAsd(
+            [pc_vel, pc_acc], path, parametrizer="ParametrizeConstAccel"
+        )
+        instance.set_desired_duration(duration)
```

Comment:
seems to me this is the only change. would you mind just modifying the TOPP function above and add `duration=None` as default argument?

Reply:
Yes, that's correct.
Though you might want to know that TOPPRAsd is  a bit less efficient than TOPRRA.
See: https://github.com/hungpham2511/toppra/blob/730aa8ad6854c7da01304ad1ed8c0819400862f4/toppra/algorithm/reachabilitybased/desired_duration_algorithm.py#L20
If that is not a concern for you, I'm happy to make the change.

Reply:
Ah, nevermind. I'll do it with a conditional.